modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -275,17 +279,25 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
   def addSequenceToQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] =
     addSequencesToQueue(q, qid, List(seqId))
 
-  private def removeSeq(qid: QueueId, seqId: Observation.Id): Endo[EngineState] = st => (
-    for {
-      q <- st.queues.get(qid)
-      if (q.queue.contains(seqId))
-      if (q.status(st) =!= BatchExecState.Running || st.executionState.sequences.get(seqId)
-        .map(seq => !seq.status.isRunning && !seq.status.isCompleted).getOrElse(true))
-    } yield queueO(qid).modify(_.removeSeq(seqId))(st)
-  ).getOrElse(st)
+  private def removeSeq(qid: QueueId, seqId: Observation.Id): executeEngine.HandleType[Unit] =
+    executeEngine.get.flatMap{ st => (
+      for {
+        q <- st.queues.get(qid)
+        if (q.queue.contains(seqId))
+        sstOp = st.executionState.sequences.get(seqId).map(_.status)
+        if (q.status(st) =!= BatchExecState.Running ||
+          sstOp.map(sst => !sst.isRunning && !sst.isCompleted).getOrElse(true))
+      } yield executeEngine.modify(queueO(qid).modify(_.removeSeq(seqId))) *>
+        ((q.cmdState, q.status(st)) match {
+          case (_, BatchExecState.Completed)         => executeEngine.unit
+          case (BatchCommandState.Run(o, u, c), _)
+            if sstOp.map(_.isError).getOrElse(false) => runQueue(qid, o, u, c)
+          case _                                     => executeEngine.unit
+        })
+    ).getOrElse(executeEngine.unit)}
 
   def removeSequenceFromQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes]((removeSeq(qid, seqId) withEvent UpdateQueueRemove(qid, List(seqId))).toHandle)
+    Event.modifyState[executeEngine.ConcreteTypes](removeSeq(qid, seqId).map[executeEngine.ConcreteTypes#EventData](_ => UpdateQueueRemove(qid, List(seqId))))
```

Comment:
Why this?

Reply:
Because I changed `removeSeq` to produce a `Handle[Unit]` instead of a `Endo`, but I still need to output the `SeqEvent`, so I added it at the end with the `map`.

Reply:
And the reason why I changed `removeSeq`, was to be able to start sequences inside with the call to `runQueue`.